src/components/Cart/BillingForm.tsx

```diff
@@ -0,0 +1,168 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Controller, useForm } from "react-hook-form";
+import { billingValidationShema } from "./validation";
+import FormField from "@components/FormComponents/FormField";
+import { Button } from "@components/Button/Button";
+import { ButtonVariant } from "../../main/types/enums";
+import { Autocomplete } from "@components/FormComponents/Autocomplete";
+import { FormControl } from "@mui/material";
+import { twMerge } from "tailwind-merge";
+import Label from "@components/FormComponents/Label";
+import { useEffect, useMemo } from "react";
+import { AppDispatch } from "../../redux/app/store";
+import { useDispatch } from "react-redux";
+import {
+  getCities,
+  getCountries,
+} from "../../redux/features/location/locationSlice";
+import { useAppSelector } from "../../redux/app/hooks";
+import {
+  transformCitiesToOptions,
+  transformCountriesToOptions,
+} from "../../main/helpers";
+import {
+  selectCities,
+  selectCountries,
+} from "../../redux/features/location/selectors";
+
+export const BillingForm = () => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    control,
+    watch,
+    setValue,
+  } = useForm({
+    resolver: yupResolver(billingValidationShema),
+  });
+
+  const country = watch("country");
+  const dispatch: AppDispatch = useDispatch();
+  const countries = useAppSelector(selectCountries);
+  const cities = useAppSelector(selectCities);
+
+  useEffect(() => {
+    dispatch(getCountries());
+  }, []);
+
+  useEffect(() => {
+    if (country) {
+      dispatch(getCities({ country: country }));
+    } else {
+      setValue("city", "");
```

Comment:
I think it is better to use `resetField` here

Reply:
done